### PR TITLE
Chaos

### PR DESCRIPTION
**Pull request by MattCruikshank**
Please review, this doesn't work without the config change!

### CODE REVIEW

File: demo_modules/chaos/chaos.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// http://jsfiddle.net/TDzVE/ - packed circles
+// http://jsfiddle.net/MattCruikshank/Aq3z8/16/ - chaos theory
+
+/* Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Copyright has to come first.

File: package.json [hunk 2]
```diff
@@ -29,8 +29,8 @@
     "relaxed-json": "^1.0.0",
     "request": "^2.64.0",
     "requirejs": "^2.1.19",
-    "socket.io": "^1.3.5",
-    "socket.io-client": "^1.3.5",
+    "socket.io": "1.3.5",
```
Comment (applmak) [inline package.json]: Probably drop this.. the socket.io issue was fixed @ Head.

**Comment by MattCruikshank** [comment]
I signed it!

File: demo_modules/chaos/chaos.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+const geometry = require('lib/geometry');
+const Rectangle = require('lib/rectangle');
+
+const network = require('network');
+const wallGeometry = require('wallGeometry');
+
+var GOOGLE_COLORS = ['#3369e8', '#d50f25', '#eeb211', '#009925'];
+
+class ChaosServer extends ModuleInterface.Server {
+  willBeShownSoon(container, deadline) {
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Elide

File: demo_modules/chaos/chaos.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  tick(time, delta) {
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Elide

File: demo_modules/chaos/chaos.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  }
+}
+
+class ChaosClient extends ModuleInterface.Client {
+  constructor(config) {
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Elide

File: demo_modules/chaos/chaos.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  }  
+
+  willBeShownSoon(container, deadline) {
+    const CanvasSurface = require('client/surface/canvas_surface');
+    this.surface = new CanvasSurface(container, wallGeometry);
+    this.canvas = this.surface.context;
+    this.canvas.fillStyle = 'black';
+    this.canvas.fillRect(0, 0, this.surface.virtualRect.w, this.surface.virtualRect.h);
+    this.imageData = this.canvas.getImageData(0, 0, this.surface.virtualRect.w, this.surface.virtualRect.h);
+    this.raw = new Array(this.surface.virtualRect.w * this.surface.virtualRect.h).fill(0);
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Optional: You *might* be happier with TypedArrays https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray

File: demo_modules/chaos/chaos.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      for (var x = 0; x < this.surface.virtualRect.w; ++x) {
+        this.imageData.data[i++] = 0;
+        this.imageData.data[i++] = 0;
+        this.imageData.data[i++] = 0;
+        this.imageData.data[i++] = 255;
+      }
+    }
+
+    this.points = 3 + 2 * Math.floor((Math.random() * 3));
+    this.alpha = Math.random() * 360.0;
```
Comment (applmak) [inline demo_modules/chaos/chaos.js]: Why 360.0?

**Pull request closed without merging**